Added unit tests and updated handle_stream_activity

vatsalkeshav opened the pull request:
Added unit tests for `handle_stream_activity`, `run_command_loop`, `calculate_new_deadline`
Also added a condition to check if current_deadline is less than timeouts.absolute_deadline to see that the deadline is not updated if it is already at the absolute limit.

cfsmp3 commented:
Can you rebase? I just pushed a cargo fmt (because your PR had it too and it made impossible to review)

vatsalkeshav commented:
done

cfsmp3 commented:
I must be blind, I don't see any change in behavior here :-)

vats004 commented:
I can't question your judgment but
before -

vats004 commented:
You're right 😅
I apologize for the misinterpretation 
but it has a good side too - now we're first checking if the deadlines are unequal and then updating 
before we were first logging"deadline unchanged " and then return

Commit:
added tests for handle_stream_activity, run_command_loop, calculate_new_deadline

Commit:
fix handle_stream_activity to pass test_handle_stream_activity_no_update_at_absolute_limit

vatsalkeshav commented:
Hi @cfsmp3 
Original deadline calc logic is perfect
So I added tests for `handle_stream_activity`, `run_command_loop`, `calculate_new_deadline`

vatsalkeshav commented:
Also fixed `handle_stream_activity` to pass `test_handle_stream_activity_no_update_at_absolute_limit`

Added a condition to check if `current_deadline` is less than `timeouts.absolute_deadline` to see that the deadline is not updated if it is already at the absolute limit.
